docs(utils/sessions): Correction about `commitSession` in non-cookie sessions (#9445)
remix-run/remix: docs/utils/sessions.md

The main advantage of cookie session storage is that you don't need any additional backend services or databases to use it. It can also be beneficial in some load-balanced scenarios. However, cookie-based sessions may not exceed the browser's max-allowed cookie length (typically 4kb).

> The downside is that you have to `commitSession` and send a "Set-Cookie" header from every loader and action that changes the session. That mean, for example, that if you `session.flash` in an action, and then `session.get` in another, you must commit it for that flashed message to go away.

Author: ```suggestion
The downside is that you have to `commitSession` and send a "Set-Cookie" header from every loader and action that changes the session. That means, for example, that if you `session.flash` in an action, and then `session.get` in another, you must commit it for that flashed message to go away.
```

> The downside is that you have to `commitSession` and send a "Set-Cookie" header from every loader and action that changes the session. That means, for example, that if you `session.flash` in an action, and then `session.get` in another, you must commit it for that flashed message to go away.

Author: ```suggestion
The downside is that you have to `commitSession` and send a "Set-Cookie" header from every loader and action that changes the session. This means, for example, that if you `session.flash` in an action, and then `session.get` in another, you must commit it for that flashed message to go away.
```

> This can cause complications when handling simultaneous requests for the same session. 

Author: ```suggestion
This can cause complications if loaders or actions are writing to the same session at the same time.
```

> With other session storage strategies you only have to send a "Set-Cookie" header when it's created (the browser cookie doesn't need to change because it doesn't store the session data, just the key to find it elsewhere). 

Author: ```suggestion
With other session storage strategies you only have to send a "Set-Cookie" header when the session is created (the browser cookie doesn't need to change because it doesn't store the session data, just the key to find it elsewhere). 
```

> Note that you still need to call `commitSession()` whenever you change the session for anything based on `createSessionStorage`, you just don't need to send an updated header.

Author: ```suggestion
Note that you still need to call `commitSession()` when you change the session for anything based on `createSessionStorage`, you just don't need to send an updated header.
```

> The downside is that you have to `commitSession` and send a "Set-Cookie" header from every loader and action that changes the session. This means, for example, that if you `session.flash` in an action, and then `session.get` in another, you must commit it for that flashed message to go away.
> 
> This can cause complications if loaders or actions are writing to the same session at the same time.
> 
> With other session storage strategies you only have to send a "Set-Cookie" header when the session is created (the browser cookie doesn't need to change because it doesn't store the session data, just the key to find it elsewhere). 
> 
> Note that you still need to call `commitSession()` when you change the session for anything based on `createSessionStorage`, you just don't need to send an updated header.

Reviewer: I think the confusion in the original docs is that the phrase "commit" is being used when it really means "set header".  So we mean to say "with others you don't need to set the header every time" but we instead incorrectly imply that with others scenarios "you don't need to commitSession every time".

What about this slight update to the original wording?  Instead of re-iterating that you need to `commitSession` after changes (which is only needed because of the incorrect implication that you don't) - we can just remove the misleading implication otherwise and indicate the true difference which relates to the `Set-Cookie` header:

```suggestion
The downside is that you have to update the cookie via a `Set-Cookie` header in almost every loader and action. If your loader or action changes the session at all, it must be sent back as an updated cookie. That means if you `session.flash` in an action, and then `session.get` in another, you must update it for that flashed message to go away. With other session storage strategies you only have to send the `Set-Cookie` header it when it's created (the browser cookie doesn't need to change because it doesn't store the session data, just the key to find it elsewhere).
```

The main advantage of cookie session storage is that you don't need any additional backend services or databases to use it. It can also be beneficial in some load-balanced scenarios. However, cookie-based sessions may not exceed the browser's max-allowed cookie length (typically 4kb).

> The downside is that you have to send a "Set-Cookie" header from almost every loader and action, this can cause complications when writing to simultaneous requests for the same session. With other session storage strategies you only have to send a "Set-Cookie" header when it's created (the browser cookie doesn't need to change because it doesn't store the session data, just the key to find it elsewhere). Note that you still need to call `commitSession()` whenever you change the session for anything based on `createSessionStorage`, you just don't need to send an updated header.

Reviewer 2: Made a mix of old & new version, and added some line feeds to make it easier to read. What do you think?

```suggestion
The downside is that you have to `commitSession` and send a "Set-Cookie" header from every loader and action that changes the session. That means if you `session.flash` in an action, and then `session.get` in another, you must commit it for that flashed message to go away.

This can cause complications when handling simultaneous requests for the same session. 

With other session storage strategies you only have to send a "Set-Cookie" header when it's created (the browser cookie doesn't need to change because it doesn't store the session data, just the key to find it elsewhere). 

Note that you still need to call `commitSession()` whenever you change the session for anything based on `createSessionStorage`, you just don't need to send an updated header.
```

Author: Great thanks

“that mean” should probably be “that means”

> The downside is that you have to send a "Set-Cookie" header from almost every loader and action, this can cause complications when writing to simultaneous requests for the same session. With other session storage strategies you only have to send a "Set-Cookie" header it when it's created (the browser cookie doesn't need to change because it doesn't store the session data, just the key to find it elsewhere). Note that you still need to call `commitSession()` whenever you change the session for anything based on `createSessionStorage`, you just don't need to send an updated header.

Author: ```suggestion
The downside is that you have to send a "Set-Cookie" header from almost every loader and action, this can cause complications when writing to simultaneous requests for the same session. With other session storage strategies you only have to send a "Set-Cookie" header when it's created (the browser cookie doesn't need to change because it doesn't store the session data, just the key to find it elsewhere). Note that you still need to call `commitSession()` whenever you change the session for anything based on `createSessionStorage`, you just don't need to send an updated header.
```
